### PR TITLE
HRQB 52 - Salary History Updates

### PR DESCRIPTION
**Pull request by ghukill**
### Purpose and background context

This PR updates the transformation and columns loaded to the `Employee Salary History` table.  Updates include:

- removing any rows from the data warehouse where the salary adjustment was a "Salary Supplement" ([commit](https://github.com/MITLibraries/hrqb-client/commit/a04723162bbc0e9dbecb11a7942b19889bd84043))
- rename a field ([commit](https://github.com/MITLibraries/hrqb-client/commit/4cb23e4efc13fda3a690a1297c3403375c46aeff))
- add a new field `Effective Salary Change %` ([commit](https://github.com/MITLibraries/hrqb-client/commit/8b241f6ac6610a01755a50cfafdbd1b78ec8876c))

### How can a reviewer manually see the effects of these changes?

Not possible to test manually without persmissions.

The changes are mostly naming, though the new test `test_task_transform_employee_salary_history_set_effective_change_percent_with_temps()` shows testing of calculations for the change in effective salary.

### Includes new or updated dependencies?
NO

### Changes expectations for external applications?
YES: new columns written to Quickbase

### What are the relevant tickets?
- https://mitlibraries.atlassian.net/browse/HRQB-52

### Developer
- [X] All new ENV is documented in README
- [X] All new ENV has been added to staging and production environments
- [X] All related Jira tickets are linked in commit message(s)
- [X] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer(s)
- [ ] The commit message is clear and follows our guidelines (not just this PR message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The provided documentation is sufficient for understanding any new functionality introduced
- [ ] Any manual tests have been performed **or** provided examples verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: hrqb/tasks/employee_salary_history.py [hunk 1]
```diff
@@ -144,13 +151,51 @@ def _set_base_salary_change_percent(self, salary_df: pd.DataFrame) -> pd.DataFra
...
             ),
-            3,
+            PERCENT_DECIMAL_ACCURACY,
         )
         new_salary_df["base_change_percent"] = new_salary_df["base_change_percent"].where(
             new_salary_df["previous_base_amount"].notna(), 0.0
         )
         return new_salary_df
 
+    def _set_effective_salary_and_change_percent(
```
Comment (ghukill) [inline hrqb/tasks/employee_salary_history.py:161]: This is _very_ similar to the method `_set_base_salary_change_percent` defined above: calculating a change percentage based on previous rows.  

It _almost_ lends itself to abstracting out to a shared method where you pass fields in the dataframe... but likely only these two, so probably too much overhead for the gain.

File: hrqb/tasks/employee_salary_history.py [hunk 2]
```diff
@@ -144,13 +151,51 @@ def _set_base_salary_change_percent(self, salary_df: pd.DataFrame) -> pd.DataFra
...
+        self, salary_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        new_salary_df = salary_df.copy()
+
+        # set effective salary to a temp base, or use the original base amount
+        new_salary_df["effective_salary"] = (
+            new_salary_df["temp_change_base_amount"]
+            .replace(0, np.nan)
```
Comment (jonavellecuerdo) [inline hrqb/tasks/employee_salary_history.py:169]: Is the following true: If `new_salary_df["temp_change_base_amount"] == 0`, then `original base amount`?

File: tests/tasks/test_employee_salary_history.py [hunk 3]
```diff
@@ -108,9 +108,105 @@ def test_task_transform_employee_salary_history_set_base_change_percent(
...
-            0.456,
+            0.45631,  # 5 digits of accuracy, but rounded to 2 digits in Quickbase
             0.0,  # new position, so no change
             0.125,
             0.1,
         ],
     )
+
+
+def test_task_transform_employee_salary_history_set_effective_change_percent_no_temps(
```
Comment (jonavellecuerdo) [inline tests/tasks/test_employee_salary_history.py:119]: Hmm, for completeness, can we also include an assertion for the `effective_salary` column?

**Review by jonavellecuerdo — CHANGES_REQUESTED** [review]
Looking good! Just one small change request for the added tests. 🤓

File: hrqb/tasks/employee_salary_history.py [hunk 4]
```diff
@@ -12,6 +13,8 @@
 from hrqb.tasks.employee_appointments import TransformEmployeeAppointments
 from hrqb.utils import md5_hash_from_values, normalize_dataframe_dates
 
+PERCENT_DECIMAL_ACCURACY = 5
```
Comment (ehanson8) [inline hrqb/tasks/employee_salary_history.py:16]: Great use of a constant!

**Review by ehanson8 — APPROVED** [review]
Approved after @jonavellecuerdo's requested change!

File: hrqb/tasks/employee_salary_history.py (hunk 2 above)
Comment (ghukill) [reply hrqb/tasks/employee_salary_history.py:169]: That's correct!

File: tests/tasks/test_employee_salary_history.py (hunk 3 above)
Comment (ghukill) [reply tests/tasks/test_employee_salary_history.py:119]: I see you're thinking here @jonavellecuerdo, but the values for the `effective_salary`actually _are_ tested as part of this test here: https://github.com/MITLibraries/hrqb-client/pull/177/files#diff-260cf8b13f4efa8a193455f2e744008f6124140071150436a72720eb9f0c4e9bR192-R201

The `effective_salary` values -- while needed to calculate the change percentage -- are not actually written to Quickbase (it's a "formula" field there), hence not having a dedicated test.

Are you comfortable with how this test _does_ check the intermediate `effective_salary` values?  It's consistent with the [pre-existing test](https://github.com/MITLibraries/hrqb-client/pull/177/files#diff-260cf8b13f4efa8a193455f2e744008f6124140071150436a72720eb9f0c4e9bR143-R164) that checks first the value being considered, and then the calculated percentage change.

Comment (jonavellecuerdo) [reply tests/tasks/test_employee_salary_history.py:119]: I think I follow-- if the assertions on the `previous_effective_salary` and `effective_change_percent` pass, then it's safe to assume `effective_salary` are the values you expected. I'm prolly just not quick with back-calculating percent changes 😅 , but I think I'm okay without the changes~

```
np.testing.assert_array_equal(
        new_df.effective_salary.values,
        [
            10_000,
            10_300,
            15_000,
            20_000,
            22_500,
            24_750
        ],
    )
```

**Review by jonavellecuerdo — APPROVED** [review]

File: tests/tasks/test_employee_salary_history.py (hunk 3 above)
Comment (ghukill) [reply tests/tasks/test_employee_salary_history.py:119]: Merged, but just to close the loop and respond.

Ahhhh.... just deleted my response. I see your point!  You were talking about asserting that the `effective_salary` values are what we expect, specifically testing this logic:

```python
new_salary_df["effective_salary"] = (
    new_salary_df["temp_change_base_amount"]
    .replace(0, np.nan)
    .fillna(new_salary_df["original_base_amount"])
)
```

That's straight-forward enough, I don't think it needs testing.  If it's 0 (zero), then we replace with `np.nan`, and then followup that by filling `np.nan` with the `original_base_amount`. 

We kind of _indirectly_ check them in the code I shared above, by checking the "previous" form of each, which is just one row behind.  

I take your point though: I'll make a note to revisit these tests.  Want to get this deployed for review, but they would benefit from another pass.

**Pull request merged**